pkg/config/labels/instance.go

```diff
@@ -92,14 +92,20 @@ func (i Instance) Validate() error {
 
 // IsDNS1123Label tests for a string that conforms to the definition of a label in
 // DNS (RFC 1123).
-func IsDNS1123Label(value string) bool {
-	return len(value) <= DNS1123LabelMaxLength && dns1123LabelRegexp.MatchString(value)
+func IsDNS1123Label(value string) (bool, error) {
+	if len(value) <= DNS1123LabelMaxLength && dns1123LabelRegexp.MatchString(value) {
```

Comment:
The error message(s) would be more clear if you break this in two -- check for DNS1123LabelMaxLength and report if there is a problem, check for MatchString and report that problem separately.